### DISCUSSION TITLE
Attribution (copyright)

### DISCUSSION THREAD

**Discussion by Jip-Hop** [Jip-Hop/jailmaker#181] [Announcements]
I'm planning to release Jailmaker under the [LGPL-3.0 license to make it compatible with the TrueNAS Middleware repo](https://github.com/truenas/middleware/blob/master/LICENSE). I've put `__author__ = "Jip-Hop"` at the top of `jlmkr.py`. If you want to be listed as author for to code you've added to this file please add your name and provide a pull request on the develop branch. @TrueJournals @templehasfallen @maxnordlund @dalgibbard @qudiqudi @bcat @HeavenVolkoff

**Comment by HeavenVolkoff** [comment]
[My contribution was very small](https://github.com/Jip-Hop/jailmaker/pull/11), so I don't think my name should be listed as author. Thanks for the opportunity though. Also, nice work maintaining this project.

**Comment by qudiqudi** [comment]
I too only contributed minor stuff. I wouldn't want claim authorship for that. I am really glad to have helped this project in the early days. Thank you @Jip-Hop for all the hard work.

**Comment by dalgibbard** [comment]
It's all good dude, just happy to help where I can; no need for specific attribution

**Comment by bcat** [comment]
My one PR was also quite small. I don't feel the need to be listed separately, but for posterity: I agree to the LGPL relicensing.

Congrats on getting this upstreamed! I hope this is the start of a deeper integration with TrueNAS itself. :)

**Comment by Lockszmith-GH** [comment]
@Jip-Hop, I know you're proud of this project - we all here feel the amount of time and effort you've put into it.
And making it LGPL is **amazing**, 

As you've brought something that was 'already there' to the front, making it easy to use.
You've done so not by sheer technical skill, but with the ability to create a vibrant, active community around it. That is probably much harder than the technical experties, and the combination of successfully doing both is a rare skill that is very hard to teach/transfer - probably the main difference between active/live projects and dead/stale ones.

- May I suggest you do it the way GitHub shows contributors - add anyone, small or large at the by-line (maybe even by automation).
Anyone you choose to add there, is high praise to the individual community members who just wanted to share this and make it better.

Awesome work!

**Reply by Jip-Hop** [reply]
Thanks for the very kind words! And I am grateful for all contributions, including non-code contributions like yours. For now I'm leaving it up to individual contributors to decide if they want to be listed in the code as author. Of course there's always this: https://github.com/Jip-Hop/jailmaker/graphs/contributors 🙂

**Reply by Lockszmith-GH** [reply]
That's exactly what I mean. Trust the system :)

**Comment by maxnordlund** [comment]
I'm good, thanks. For posterity I agree to relicense my contribution as LGPL.

**Comment by templehasfallen** [comment]
That's great news on the license and roadmap. I'm happy to be listed and
hope i'll be able to contribute more in the near future.

**Comment by CompyENG** [comment]
Sorry for the delayed response, but all good from me as well.  I've got not problem with just showing up in the git history.  Thanks!

**Comment by jonct** [comment]
I'm late to the game, but researching how to best reinforce and [express](https://spdx.dev/about/overview/) Jip-Hop's decision through metadata.

[Matija "Hook" Šuklje](https://matija.suklje.name) wrote an extraordinary if not definitive article "[How and why to properly write copyright statements in your code](https://matija.suklje.name/how-and-why-to-properly-write-copyright-statements-in-your-code)".

On the basis of that article, ⚖️ I've started putting the following at the top of every new file containing code or build configuration.
```
# SPDX-FileCopyrightText: © 2024 Jip-Hop and the Jailmakers <https://github.com/Jip-Hop/jailmaker>
#
# SPDX-License-Identifier: LGPL-3.0-only
```
If you find it too cheeky, let me know what to use instead — but please do thoughtfully take in the article first. Cheers!
